src/OrchardCore.Modules/OrchardCore.Users/TimeZone/Services/UserTimeZoneService.cs

```diff
@@ -8,77 +8,80 @@
 {
-    private const string CacheKey = "UserTimeZone/";
-
-    private readonly TimeSpan _slidingExpiration = TimeSpan.FromMinutes(1);
-    private readonly IClock _clock;
-    private readonly IDistributedCache _distributedCache;
-    private readonly IHttpContextAccessor _httpContextAccessor;
-    private readonly UserManager<IUser> _userManager;
-
-    public UserTimeZoneService(
-        IClock clock,
-        IDistributedCache distributedCache,
-        IHttpContextAccessor httpContextAccessor,
-        UserManager<IUser> userManager)
+    public class UserTimeZoneService
     {
-        _clock = clock;
-        _distributedCache = distributedCache;
-        _httpContextAccessor = httpContextAccessor;
-        _userManager = userManager;
-    }
-
-    public async Task<ITimeZone> GetUserTimeZoneAsync()
-    {
-        var currentTimeZoneId = await GetCurrentUserTimeZoneIdAsync();
-
-        if (string.IsNullOrEmpty(currentTimeZoneId))
+        private const string CacheKey = "UserTimeZone/";
+        private readonly TimeSpan _slidingExpiration = TimeSpan.FromMinutes(1);
+
+        private readonly IClock _clock;
+        private readonly IDistributedCache _distributedCache;
+        private readonly IHttpContextAccessor _httpContextAccessor;
+        private readonly UserManager<IUser> _userManager;
+
+        public UserTimeZoneService(
+            IClock clock,
+            IDistributedCache distributedCache,
+            IHttpContextAccessor httpContextAccessor,
+            UserManager<IUser> userManager
+            )
         {
-            return null;
+            _clock = clock;
+            _distributedCache = distributedCache;
+            _httpContextAccessor = httpContextAccessor;
+            _userManager = userManager;
         }
 
-        return _clock.GetTimeZone(currentTimeZoneId);
-    }
+        public async Task<ITimeZone> GetUserTimeZoneAsync()
+        {
+            var currentTimeZoneId = await GetCurrentUserTimeZoneIdAsync();
+            if (string.IsNullOrEmpty(currentTimeZoneId))
+            {
+                return null;
+            }
 
-    public Task ClearUserTimeZoneCacheAsync(User user)
-    {
-        var userName = user?.UserName;
+            return _clock.GetTimeZone(currentTimeZoneId);
+        }
 
-        return string.IsNullOrEmpty(userName)
-            ? Task.CompletedTask
-            : _distributedCache.RemoveAsync(GetCacheKey(userName));
-    }
+        public Task UpdateUserTimeZoneAsync(User user)
+        {
+            var userName = user?.UserName;
 
-    public async Task<string> GetCurrentUserTimeZoneIdAsync()
-    {
-        var userName = _httpContextAccessor.HttpContext.User?.Identity?.Name;
+            if (!string.IsNullOrEmpty(userName))
+            {
+                return _distributedCache.RemoveAsync(GetCacheKey(userName));
+            }
 
-        if (string.IsNullOrEmpty(userName))
-        {
-            return null;
+            return Task.CompletedTask;
```

Comment:
The original ternary was more succint.

Reply:
I'm sorry but how did you end up commenting on file changes that we're removed from this PR?
I've put the TimeZone changes in a different PR.

Reply:
I commented on the diff that GitHub showed since my last review.